src/main/java/com/google/netpcapanalysis/dao/MaliciousIPDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.interfaces.dao.MaliciousIPDao;
+import com.google.netpcapanalysis.models.PCAPdata;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import java.io.IOException;
+
+import javax.lang.model.util.ElementScanner6;
+
+public class  MaliciousIPDaoImpl implements MaliciousIPDao{
+    private static final String AUTH0_API_KEY = "replace_api_key";
+
+    private static final String FLAGGED_FALSE = "Resource not found";
+    private static final String FLAGGED_TRUE = "200: OK";
+    private static final String REQUEST_LIMIT = "Rate limit exceeded.Please reduce your hits per minute.";
+
+    public MaliciousIPDaoImpl(){
+
+    }
+    public PCAPdata isMalicious(PCAPdata data)
+    {
+        HttpResponse<String> result;
+        try {
+            result = Unirest.get("https://signals.api.auth0.com/badip/" + data.destination)
```

Comment:
Future: You could create a abstraction of this that will help you avoid calling this API and just return if you have already hit the limit. Or build a simple cache and lookup first.